### ISSUE TITLE
commandline name

### ISSUE THREAD

**Issue by Memnarch** [DelphiPackageManager/DPM#8]
Hi,
Seems you are working on a package manager aswell. Is it possible to use another acronym then "DPM"? Delphinus has been using this name for some years now for the commandline and yours might collide, if people add these tools to their path-variable.

PS: I'm wondering, why did you start your own project, instead of joining Delphinus?

**Comment by vincentparrett** [comment]
I'm happy to change, I just don't have a better name right now. Open to suggestions. 

I started this project back in 2013 (I shelved it when getit was announced, but getit has been a total disappointment) . 

I did look at delphinus, but I find the code hard to follow, and I didn't like the way it works - installed packages globaly vs per project, tied to github (so no offline or third party feeds etc).  

Nothing personal, I just wanted something that works how I want it to work.

**Comment by Memnarch** [comment]
Currently implementing per project support
https://github.com/Memnarch/Delphinus/tree/feature/projectbasedsetup
It is not tied to github. Github is just the only source implemented atm. But the mechanics don't rely on Github being the backbone. I haven't added support for multiple sources or configuring them, yet.

In short, all it needs is to implement IDNPackageProvider.
https://github.com/Memnarch/Delphinus/blob/master/DN.PackageProvider.Intf.pas

Anything i can help you with in understanding the source?

And no worries, I don't take this personal. The only thing I'm worried is that 2 people put an imense amount of work into similar things.

**Comment by vincentparrett** [comment]
I guess we just have different idea's on how packages should be managed. I work a lot with .net and javascript and wanted something closer to nuget and npm/yarn. 

I evaluated DN (when deciding whether to continue my project),  it was still tied to github and it couldn't work offline.  That was a deal breaker for me, I need a package manager that will work from a CI server, using a local package feed so I can still build even when offline. 

I have a question for you though, why did you name your command line dpm.exe instead of dn.exe (which seems like the more obvious choice). Any chance you could change it? As I'm sure you know, naming is the hardest thing when creating software projects ;) I did try looking for a new name, but it's hard to change when it's been the same since 2013 (although I only recently published it) and I have not found a better or good alternative name.

**Comment by Memnarch** [comment]
That's actually a thing i never understood:
You needed support for an offline feed, than why didn't you just help with Delphinus, instead of reworking an entire packagemanager around it? After all, a custom feed is just one implementation away.

Before i went dark, i started working on allowing n-sources. With their own abstraction to store their needed configurations for UI and commandline, too
Which is this branch:
https://github.com/Memnarch/Delphinus/tree/feature/MultiSource

Until recently, i had not much time to work on all of this, but hope to have again.

Oh and regarding DN, that is some obscure Namespace-Decision i made back then, when i started working on the project. It didn't age well to be honest.

**Comment by vincentparrett** [comment]
> That's actually a thing i never understood:
You needed support for an offline feed, than why didn't you just help with Delphinus, instead of reworking an entire packagemanager around it? After all, a custom feed is just one implementation away.

As I said, I started working on this along time ago, and modelled it after nuget and npm, when I looked at DN it was a long way from what I was looking for. 

> Oh and regarding DN, that is some obscure Namespace-Decision i made back then, when i started working on the project. It didn't age well to be honest.

Naming is hard!

**Comment by Memnarch** [comment]
> Naming is hard!
I hope you understand that I won't rename my tool as it has been in the wild for some time now.
Regarding the name of yours: Open for anything that doesn't start with Delphi right away? No idea yet, but depending on the name, the acronym for your commandline might work right away.

**Comment by vincentparrett** [comment]
> I hope you understand that I won't rename my tool as it has been in the wild for some time now.
Just to be clear, I wasn't suggesting you should rename delphinus, just the command line tool.
dn.exe makes more sense for that than dpm.exe, tbh I was a bit surprised to find your cmd line name was dpm. 

I'm going to leave dpm as is for now, at least until a better name arises, rather than pick any crappy name for the sake of changing.

**Comment by Memnarch** [comment]
Back then I picked DPM for "Delphinus Paket Manager" like Nuget has NPM :P
